### DISCUSSION TITLE
Ignored words list

### DISCUSSION THREAD

**Discussion by sajadhsm** [sajadhsm/new-word-tab#21] [Ideas]
Based on discussions (#14 & [#15](https://github.com/sajadhsm/new-word-tab/issues/15#issuecomment-1052006978)) by @silverwings15 having an ignored words list seems to be useful.

It enables us to:
- Separate words which don't return definition from learned words
- Add common words to the ignored list so learned words become much handier to review

By introducing another word list, we also need a new word action button. Since the actions section is already full of buttons, I think it's a good idea to change the UI according to the image below.

![Actions](https://user-images.githubusercontent.com/20022818/155850807-813742b0-897d-4c66-a8d4-13444390ed9a.png)

To keep things simple, it may be better to remove the recently added ["Change Action Buttons Position Ability"](https://github.com/sajadhsm/new-word-tab/pull/19). Because in the future, we probably add more actions and then we need to consider how the actions look in each position. I think the new UI is even better than all current positioning options too.

Now that we've separated the ignored words from the learned ones, it's better to move the learned words list to the main page so it's easier the access them.

Based on the image below, the marked words drawer will have 2 tabs for "Marked" and "Learned" words. The default tab will be set on Marked words since they are viewed more often.

![Marked and Learned words](https://user-images.githubusercontent.com/20022818/155850783-e1a4d03f-9cc1-4dd9-8639-7be56c6c395a.png)

Ignored words are kept in Options pages since we rarely need to visit them.

![Ignored Words](https://user-images.githubusercontent.com/20022818/155850805-6d1d58f6-3be3-4e32-abe0-31deb0bf33e8.png)

Sorting and filtering will be added for the ignored list too. And to keep the options page look cleaner maybe the ignored words section will be collapsed by default.

**Comment by silverwings15** [comment]
you hand drew all this? awesome! and i agree with everything you wrote.

one question though, what do you have in mind for the sorting/filtering functions? search bar for filtering and sorting by list/alphabetical order/date added etc?

**Reply by sajadhsm** [reply]
Yeah, did some quick wireframes :D

Exactly. Similar to what we currently have for the Marked words list.

**Reply by silverwings15** [reply]
oh haha i just realized i never accessed the Marked words section. yeah looks good for sure, and Learned/Ignored lists will definitely benefit from having the same filters.

**Comment by sajadhsm** [comment]
Hi @silverwings15 
Finally, after a long absence, I found some time to implement this feature. :D

[Changelong for v1.5.0](https://github.com/sajadhsm/new-word-tab/releases/tag/v1.5.0)

The ignored list also has an import file button which allows the user to upload a list of words.
With the import file components implemented, we are one step closer to having the ability to define custom vocabulary lists :)

**Reply by silverwings15** [reply]
hello, glad you're well. i also just went through a super busy period, can now hopefully resume working on those word lists we discussed before

**Reply by silverwings15** [reply]
the "bottleneck" now i think is that there hasn't been any progress on the freedictionaryAPI for some time, so there are still a lot of words that return incomplete or non-existent definitions
